Automattic/wp-calypso
# Change editor block learn more link to open support site in Help Center

## Proposed Changes

* Updates post editor block "learn more" link to open support link in Help Center instead of navigating to link in a new tab.

https://github.com/Automattic/wp-calypso/assets/10482592/6d75a294-6c36-4f4b-bce3-afe4cc1602ed

<img width="1084" alt="CleanShot 2024-01-08 at 15 21 39@2x" src="https://github.com/Automattic/wp-calypso/assets/10482592/11f43964-6a7a-4f1d-95c5-3c0101aed257">



## Testing Instructions



* Checkout branch
* Sandbox public.api and your test site
* Navigate to `apps/editing-toolkit` and sync changes to sandbox: `yarn dev --sync`
* Navigate to Post editor
* Add various blocks to your post. Select block, open the editor "settings" tab, and click the "Learn more" link.
* Verify the help center opens and displays the support link for the selected block.
* Repeat for different blocks
* Verify `devdocs/blocks/support-article-dialog` SupportArticleDialog component is still working.

## Pre-merge Checklist
